crates/rattler_solve/src/libsolv/wrapper/solvable.rs

```diff
@@ -21,60 +13,27 @@ impl From<SolvableId> for ffi::Id {
 }
 
 impl SolvableId {
-    /// Resolves to the interned type returns a Solvable
-    pub fn resolve(&self, pool: &Pool) -> Solvable {
-        // Safe because the new-type wraps the ffi::id and cant be created otherwise
-        unsafe {
-            // Re-implement pool_id2solvable, as it's a static inline function, we can't use it :(
-            let solvables = pool.as_ref().solvables;
-            // Apparently the solvable is offset by the id from the first solvable
-            let solvable = solvables.offset(self.0 as isize);
-            Solvable(
-                NonNull::new(solvable).expect("solvable cannot be null"),
-                PhantomData::default(),
-            )
-        }
-    }
-}
-
-impl<'repo> Solvable<'repo> {
-    /// Returns a pointer to the wrapped `ffi::Solvable`
-    pub(super) fn as_ptr(&self) -> NonNull<ffi::Solvable> {
-        self.0
-    }
-
-    /// Looks up a string value associated with this instance with the given `key`.
-    pub fn lookup_str(&self, key: StringId) -> Option<&str> {
-        let str = unsafe { ffi::solvable_lookup_str(self.0.as_ptr(), key.into()) };
-        if str.is_null() {
-            None
+    /// Resolves the id to a pointer to the solvable
+    pub fn resolve_raw(self, pool: &Pool) -> NonNull<ffi::Solvable> {
```

Comment:
Should this function remaning local to the crate because it returns an ffi object? (`pub(super)`)

Reply:
Also, explain that this function might panic. Similar to resolving of strings.

Reply:
I intentionally exposed a pointer to `ffi::Solvable` because we need mutable access to it in `input.rs`. The ffi object is difficult to wrap in a safe interface, because it is mostly useful when having a mutable reference to it. By exposing it as a raw pointer, the user takes on the responsibility of ensuring there is at most one reference to `ffi::Solvable` at a given time.

I have spent about 30 minutes trying to create a `Solvable` wrapper that is safer, but am not satisfied with the result. If you think it is truly worth it, I can give it a second try. Otherwise I'd suggest we leave it as is